x-pack/legacy/plugins/lens/public/indexpattern_plugin/operation_definitions/filter_ratio.tsx

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { useState } from 'react';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n/react';
+import { EuiButton, EuiFormRow } from '@elastic/eui';
+import { Query } from '../../../../../../../src/legacy/core_plugins/data/public/query';
+import { FilterRatioIndexPatternColumn } from '../indexpattern';
+import { DimensionPriority } from '../../types';
+import { OperationDefinition } from '../operations';
+import { updateColumnParam } from '../state_helpers';
+
+export const filterRatioOperation: OperationDefinition<FilterRatioIndexPatternColumn> = {
+  type: 'filter_ratio',
+  displayName: i18n.translate('xpack.lens.indexPattern.filterRatio', {
+    defaultMessage: 'Filter Ratio',
+  }),
+  isApplicableWithoutField: true,
+  isApplicableForField: () => false,
+  buildColumn(
+    operationId: string,
+    suggestedOrder?: DimensionPriority
+  ): FilterRatioIndexPatternColumn {
+    return {
+      operationId,
+      label: i18n.translate('xpack.lens.indexPattern.filterRatio', {
+        defaultMessage: 'Filter Ratio',
+      }),
+      dataType: 'number',
+      operationType: 'filter_ratio',
+      suggestedOrder,
+      isBucketed: false,
+      params: {
+        numerator: { language: 'kuery', query: '' },
+        denominator: { language: 'kuery', query: '*' },
+      },
+    };
+  },
+  toEsAggsConfig: (column, columnId) => ({
+    id: columnId,
+    enabled: true,
+    type: 'filters',
+    schema: 'segment',
+    params: {
+      filters: [
+        {
+          input: column.params.numerator,
+          label: '',
+        },
+        {
+          input: column.params.denominator,
+          label: '',
+        },
+      ],
+    },
+  }),
+  paramEditor: ({ state, setState, columnId: currentColumnId, dataPlugin, storage }) => {
+    const [hasDenominator, setDenominator] = useState(false);
+
+    if (!dataPlugin || !storage) {
+      return null;
```

Comment:
In what condition would these be unavailable, and is it an error that we should report?

Reply:
They won't be in practice, but this was added because I've made these into optional properties for testing. It's a huge pain to provide mocks for this in every dimension panel test.